bus-mapping/src/evm/opcodes/calldatacopy.rs

```diff
@@ -0,0 +1,203 @@
+use super::Opcode;
+use crate::circuit_input_builder::{CircuitInputStateRef, ExecState, ExecStep, StepAuxiliaryData};
+use crate::operation::{CallContextField, CallContextOp, RWCounter, RW};
+use crate::Error;
+use eth_types::evm_types::ProgramCounter;
+use eth_types::GethExecStep;
+use std::collections::HashMap;
+
+// The max number of bytes that can be copied in a step limited by the number
+// of cells in a step
+const MAX_COPY_BYTES: usize = 71;
+
+#[derive(Clone, Copy, Debug)]
+pub(crate) struct Calldatacopy;
+
+impl Opcode for Calldatacopy {
```

Comment:
does need to add `CopyToMemory` bus mapping implementation  as in circuit it is a step gadget ?

Reply:
It seems that `CallDataCopy` and `CopyToMemory` steps are generated from one GethExecStep. Since CopyToMemory steps are virtual. I added function `gen_memory_copy_steps` to generate CopyToMemory steps based on the previous `CallDataCopy` step.